### PR TITLE
Generate dynamic alias for join source

### PR DESCRIPTION
**Pull request by rishi-aga**
One of the PR that Resolves https://github.com/yahoo/elide/discussions/1829

## Description
Generate dynamic alias for join source

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/QueryTranslator.java [hunk 1]
```diff
@@ -282,6 +282,7 @@ private String extractOrderBy(Map<Path, Sorting.SortOrder> sortClauses, Query qu
      */
     private Set<String> extractJoinExpressions(Query query) {
         return query.getColumnProjections().stream()
+                .filter(column -> column.isProjected())
```
Comment (aklish) [inline elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/QueryTranslator.java:285]: The params above in the javadoc are wrong.  This should explain that this only extracts joins from projected columns.

File: elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+
+    @DimensionFormula("{{join.joinjoin.dim2}}")
+    private String dim3;
+
+    @Join("{{$id}} = {{join.$id}}")
+    private JoinTable join;
+
+    @DimensionFormula(value = "{{join2.dim1}}",
+                      arguments = { @ArgumentDefinition(name = "exprArg", type = ValueType.TEXT),
+                                    @ArgumentDefinition(name = "joinArg4", type = ValueType.TEXT)})
```
Comment (aklish) [inline elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java:451]: The 4 in joinArg4 is confusing.  Do the numbers mean anything?

File: elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+                        .dimensionProjection(dim4)
+                        .dimensionProjection(dim5)
+                        .arguments(emptyMap())
+                        .build();
+
+        String generatedSql = engine.explain(query).get(0);
+
+        String expectedSQL =
+                          "SELECT " + NL
+                        + "  DISTINCT `MainTable_join2_XXX`.`dim1` AS `dim4`," + NL
```
Comment (aklish) [inline elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java:235]: Is there any way we can tell the XXX aliases are different here?

File: elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java [hunk 4]
```diff
@@ -0,0 +1,541 @@
...
+                        + "  value = 'same' AND `MainTable`.`dim6` - 'bar5' = `MainTable_join2_XXX`.`dim4` - 'foo' ";
+
+        assertEquals(formatExpected(expectedSQL), formatGenerated(generatedSql));
+    }
+
+    // Both dim4a and dim5a passes same value for join expression's argument 'exprArg'
+    // dim4a and dim5a passes different value for join table's column argument 'joinArg4'
+    // 1 Join expressions is generated as fixed value of joinArg4 is used while calling join table's column.
+    @Test
+    void testArgumentsInJoinExprCase5() {
```
Comment (aklish) [inline elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java:377]: Can we add the same test but this time make exprArg vary?

File: elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java (hunk 3 above)
Comment (rishi-aga) [reply elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java:235]: using the original values.

File: elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java (hunk 2 above)
Comment (rishi-aga) [reply elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/expression/JoinExpressionExtractorTest.java:451]: renamed to joinArg

**Review by aklish — APPROVED** [review]

**Review by aklish — APPROVED** [review]

**Pull request merged**